feat(#60): 주문 조회 API 구현

joel-you opened the pull request:
## 📋 PR 요약

Order 조회 API 개발 및 Redis 캐싱 적용

### 🔗 관련 이슈
- #60

### 📝 주요 변경사항

**구현:**
- 주문 조회 API 8개 엔드포인트 (단건/목록/필터링/페이지네이션)
- Redis 캐싱 적용 (TTL 1시간)
- Response DTO 및 MapStruct Mapper 구현
- GlobalExceptionHandler 추가

**테스트:**
- 단위 테스트 14개, 통합 테스트 8개
- 모든 테스트 통과 ✅

**문서:**
- OpenAPI 3.0 스펙 작성
- Redocly 문서 생성

### 🔍 리뷰 포인트
- Redis 캐싱 전략 검토
- 에러 응답 형식 확인

joel-you committed:
refactor: List 반환 메서드 제거 및 Page 메서드로 통합

- Repository List 반환 메서드 제거, Page 메서드만 유지
- findByCustomerId, findByStatus 등 List 메서드 제거
- 모든 조회 메서드를 페이징 처리로 통합
- 테스트 코드를 Page 메서드 사용하도록 수정
- PageRequest, PageImpl 사용
- 17개 테스트 모두 통과

joel-you committed:
feat(#60): 주문 조회 API 구현

- GetOrderService 및 OrderQueryController 추가
- 단건/목록 조회, 페이징/필터링 지원

joel-you committed:
feat(#60): 주문 단건 조회에 Redis 캐싱 적용

- GetOrderService.getOrderById()에 @Cacheable 추가
- application.yml에 Redis 및 캐시 설정 추가
- build.gradle.kts에 Redis 의존성 추가
- common 모듈의 RedisConfig 활용

joel-you committed:
feat(#60): Redocly 설정 및 문서 업데이트

- redocly.yaml에 order-orchestrator@v1 API 추가

joel-you committed:
test(#60): Order 조회 API 테스트 및 예외 처리 구현

- GetOrderServiceTest: 14개 단위 테스트 (조회, 필터링, 페이지네이션)
- OrderQueryControllerIntegrationTest: 8개 통합 테스트 (REST API 검증)
- GlobalExceptionHandler: 404/400 에러 응답 처리

joel-you committed:
refactor(#60): 주문 검색 비즈니스 로직을 Service layer로 이동

- OrderQueryController에서 determineQueryMethod 및 createPageable 제거
- GetOrderService에 검색 로직 구현 (searchOrders 메서드 추가)
- GetOrderUseCase에 searchOrders 메서드 추가로 인터페이스 단순화
- Controller는 HTTP 요청 처리에만 집중하도록 단순화
- 기존 개별 조회 메서드들을 searchOrders로 통합

joel-you committed:
test(#60): 정렬 파라미터 null 처리 테스트 케이스 추가

- searchOrders_WithNullSort_AppliesDefaultSort 테스트 추가
- searchOrders_WithEmptySort_AppliesDefaultSort 테스트 추가
- searchOrders_WithWhitespaceSort_AppliesDefaultSort 테스트 추가
- 테스트 커버리지 79.43% → 80% 개선

joel-you commented on order-orchestrator/src/main/java/com/msa/commerce/orchestrator/adapter/in/web/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,46 @@
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(OrderNotFoundException.class)
+    public ResponseEntity<Map<String, Object>> handleOrderNotFoundException(
```
Map이 아닌 구현체를 명시해서 변경할 수 있도록.

joel-you replied:
ErrorResponse 라는 Class 구성해서 변경했습니다

joel-you commented on order-orchestrator/src/main/java/com/msa/commerce/orchestrator/adapter/in/web/dto/response/PageResponse.java:
```diff
@@ -0,0 +1,40 @@
+    private long totalElements;
+
+    @JsonProperty("totalPages")
+    private int totalPages;
```
다음시간에 페이징 관련된 얘기를 더 이어가도록 하는걸로

flab-true commented:
Cache를 사용하고 있는데요. 히트율이 높을까요? 어떤 관련 지표는 어떻게 수집할 수 있을까요?

flab-true commented on order-orchestrator/src/main/java/com/msa/commerce/orchestrator/adapter/in/web/OrderQueryController.java:
```diff
@@ -0,0 +1,52 @@
+
+    @GetMapping
+    public ResponseEntity<PageResponse<OrderSummaryResponse>> getOrders(OrderSearchParams searchParams) {
+        searchParams.validateAndSetDefaults();
```
`validateAndSetDefaults` 는 삭제하고 Java Bean Validation 을 고려해주세요. 
`spring-boot-starter-validation`에 대한 내부 동작 방식도 확인해 주시면 좋겠습니다.

joel-you replied:
수정했습니다

flab-true commented on order-orchestrator/src/main/java/com/msa/commerce/orchestrator/adapter/in/web/OrderQueryController.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    @GetMapping
+    public ResponseEntity<PageResponse<OrderSummaryResponse>> getOrders(OrderSearchParams searchParams) {
```
Bean Parameter 검증에서 `@Validated`, `@Valid` 차이가 무엇인지 정리해 주시면 좋겠습니다.

joel-you replied:
[wiki](https://github.com/f-lab-edu/msa-commerce-lab/wiki/Spring-Bean-Validation-%E2%80%94-%40Valid-vs-%40Validated-%EC%A0%95%EB%A6%AC) 에 기술 하였습니다.

flab-true reviewed (commented):
간단한 코멘트 남깁니다 .

joel-you committed:
#58 이슈 Merge

joel-you committed:
refactor(#67): Bean Validation 적용 및 예외 처리 개선

- Map → HashMap으로 타입 명시
- Bean Validation 적용
- 수동 검증 로직 제거
- 검증 예외 핸들러 추가

joel-you committed:
refactor: findOrdersByDateRange를 JPA 메서드 네이밍으로 변경

- @Query JPQL → JPA 메서드 네이밍 규칙 적용
- findOrdersByDateRange → findByOrderDateBetween
- Spring Data JPA 컨벤션 준수

joel-you commented:
수정완료

joel-you committed:
Remove .github/README.md from feature/60-order-retrieval-api

joel-you commented:
> Cache를 사용하고 있는데요. 히트율이 높을까요? 어떤 관련 지표는 어떻게 수집할 수 있을까요?

찾아본 방법으로는 아래와 같은 방법들이 있을 것 같습니다.
* `RedisCacheManager` 에 `enableStatistics()` 옵션을 켜놓고, `getStatistics()` 메소드를 사용해서 특정 키들에 대한 통계를 확인 할 수 있는 것 같습니다. 다만 위 옵션 자체가 약간의 성능저하가 있을 수 있다고하고, 통계치를 보기위해서, ,API를 구성하거나, 스케줄링에 의해서 Report 를 만들어내야하기 때문에, 모니터링을 위해 프로덕소스를 건들여야한다는 단점은 있는 것 같습니다.
* redis-cli 를 이용해서, 몇가지 옵션들로 Hit율 대신 모니터링 할 수 있는 지표는 있는 것으로 알고 있습니다.
  * LFU+OBJECT FREQ 로 접근빈도 확인 가능
  * MONITOR 를 이용해 실제 조회 분포를 얻고 .. 등등
* 위에서 얻을 수 있는 지표를 redis-cli로 보는 방법 보다는 그라파나/프로메테우스 같은 APM 등에서 keyspace_hits/misses를 스크랩 → 모니터링 시스템(프로메테우스/그라파나 등)에 적재해 전역 히트율 트렌드 확인할 수 있는 것으로 알고 있습니다.

첫번째로 언급한 기능을 키고, 주기적으로 Report 산출물을 만들어서 모니터링 추이를 확인하려 하였으나, 지금 당장은 비즈니스에 집중하는게 맞는 것 같아서, 개발 도중 롤백을 하게되었습니다.

joel-you committed:
refactor(#60): 예외 처리를 중앙집중식 아키텍처로 재구조화

- InternalException 추상 클래스 기반 예외 계층 구조 확립
- common의 GlobalExceptionHandler에서 모든 도메인 예외를 중앙 처리
- 각 모듈은 GlobalExceptionHandler 상속으로 예외 처리 활성화

flab-true reviewed (approved):
대단히 수고 많으셨습니다. LGTM 👍 
코드 충돌은 이후에 대응해 주세요.

joel-you committed:
Merge remote-tracking branch 'origin/develop' into feature/60-order-retrieval-api

# Conflicts:
#	order-orchestrator/src/main/java/com/msa/commerce/orchestrator/adapter/out/persistence/OrderJpaRepository.java
#	order-orchestrator/src/main/java/com/msa/commerce/orchestrator/adapter/out/persistence/OrderRepositoryImpl.java
#	order-orchestrator/src/main/java/com/msa/commerce/orchestrator/application/port/out/OrderRepository.java
#	order-orchestrator/src/test/java/com/msa/commerce/orchestrator/adapter/out/persistence/OrderRepositoryImplTest.java

joel-you committed:
refactor(#60): develop conflict 해소
